### PR TITLE
Add notification API for shared libraries

### PR DESCRIPTION
**Pull request by cRui861**
Includes a notification demo in the playground for Portal and a mock notifier (for Batch Explorer).

**Screenshots from the Portal playground demo:**
- Notification popups.
image
- Notifications in the notification tray with the pending notification is in an updated status.
image
- Notifications in the notification tray with the pending notification in a completed status.
image

**Screenshots from web Playground demo**
- Using mock notification for alerts (will replace with a Batch Explorer Notifier)
image
image

### CODE REVIEW

File: packages/common/src/ui-common/environment/environment.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { Clock } from "../datetime/clock";
 import { HttpClient } from "../http";
 import { Localizer } from "../localization";
 import type { Logger, LoggerFactory, LoggingContext } from "../logging";
+import { Notification as Notifier } from "../notification";
```
Comment (gingi) [inline packages/common/src/ui-common/environment/environment.ts:5]: This is probably a pain, but maybe it's worth renaming "Notifcation" to "Notifier" systematically to distinguish a service that provides notifications from the notifications themselves and avoid confusion in the future.

**Comment by gingi** [comment]
Rena, can you provide a screenshot or two of the demo?

**Review by gingi — CHANGES_REQUESTED** [review]
The PR looks great, but I think renaming "Notification" to "Notifier" will avoid some confusion in the future.

File: packages/common/src/ui-common/environment/environment.ts (hunk 1 above)
Comment (cRui861) [reply packages/common/src/ui-common/environment/environment.ts:5]: Yeah good point, I'll update that too.

**Review by gingi — APPROVED** [review]

**Pull request merged**